go/service/account.go

```diff
@@ -104,3 +104,65 @@ func (h *AccountHandler) ResetAccount(ctx context.Context, arg keybase1.ResetAcc
 
 	return h.G().Logout()
 }
+
+type GetLockdownResponse struct {
+	Enabled bool                       `json:"enabled"`
+	Status  libkb.AppStatus            `json:"status"`
+	History []keybase1.LockdownHistory `json:"history"`
+}
+
+func (r *GetLockdownResponse) GetAppStatus() *libkb.AppStatus {
+	return &r.Status
+}
+
+func (h *AccountHandler) GetLockdownMode(ctx context.Context, sessionID int) (ret keybase1.GetLockdownResponse, err error) {
+	apiArg := libkb.APIArg{
+		Endpoint:    "account/lockdown",
+		SessionType: libkb.APISessionTypeREQUIRED,
+	}
+	var response GetLockdownResponse
+	err = h.G().API.GetDecode(apiArg, &response)
+	if err != nil {
+		return ret, err
+	}
+
+	mctx := libkb.NewMetaContext(ctx, h.G())
+	ss, err := mctx.ActiveDevice().SyncSecrets(mctx)
+	if err != nil {
+		return ret, err
+	}
+	devs, err := ss.ActiveDevices(libkb.AllDeviceTypes)
+	if err != nil {
+		return ret, err
+	}
```

Comment:
is it better to call "list devices" engine here?

Reply:
Maybe UPAKLoader then .DeviceKeys.DeviceDescription. That's pretty well cached. I've never encountered this SyncSecrets thing it could be fine up to you.